File: decorator/java/ChocolateSauce.java
package decorator.java;

public class ChocolateSauce extends Sauce {
    public ChocolateSauce(Food food) {
        super(food);
    }

    @Override
    public void dip() {
        super.dip();
        System.out.println("Adding chocolate sauce to the food");
    }
    
}

File: decorator/golang/chocolatesauce.go
package main

type ChocolateSauce struct {
	sauce Sauce
}

func NewChocolateSauce(sauce Sauce) *ChocolateSauce {
	return &ChocolateSauce{
		sauce,
	}
}

func (c *ChocolateSauce) dip() {
	println("Adding chocolate sauce to the food and dipping it")
	c.sauce.dip()
}
